pkg/runtime/grpc/grpc-runtime.go

```diff
@@ -0,0 +1,341 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcruntime
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/kubectl-gadget/utils"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	pb "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgettracermanager/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime"
+)
+
+const (
+	GadgetServiceSocket = "/run/gadgetservice.socket"
+
+	ParamNode             = "node"
+	ParamConnectionMethod = "connection-method"
+
+	// ResultTimeout is the time in seconds we wait for a result to return from the gadget
+	// after sending a Stop command
+	ResultTimeout = 30
+)
+
+type Runtime struct {
+	catalog *runtime.Catalog
+}
+
+// New instantiates the runtime and loads the locally stored gadget catalog. If no catalog is stored locally,
+// it will try to fetch one from one of the gadget nodes and store it locally. It will issue warnings on
+// failures.
+func New() *Runtime {
+	r := &Runtime{}
+
+	// Initialize Catalog
+	catalog, err := loadLocalGadgetCatalog()
+	if err == nil {
+		r.catalog = catalog
+		return r
+	}
+
+	catalog, err = loadRemoteGadgetCatalog()
+	if err != nil {
+		log.Warnf("could not load gadget catalog from remote: %v", err)
+		return r
+	}
+	r.catalog = catalog
+
+	err = storeCatalog(catalog)
+	if err != nil {
+		log.Warnf("could not store gadget catalog: %v", err)
+	}
+
+	return r
+}
+
+func (r *Runtime) UpdateCatalog() error {
+	catalog, err := loadRemoteGadgetCatalog()
+	if err != nil {
+		return fmt.Errorf("loading remote gadget catalog: %w", err)
+	}
+
+	return storeCatalog(catalog)
+}
+
+func (r *Runtime) Init(runtimeGlobalParams *params.Params) error {
+	return nil
+}
+
+func (r *Runtime) Close() error {
+	return nil
+}
+
+func (r *Runtime) ParamDescs() params.ParamDescs {
+	return params.ParamDescs{
+		{
+			Key:         ParamNode,
+			Description: "Comma-separated list of nodes to run the gadget on",
+		},
+	}
+}
+
+func (r *Runtime) GlobalParamDescs() params.ParamDescs {
+	return params.ParamDescs{
+		{
+			Key:            ParamConnectionMethod,
+			Description:    "Method that should be used to connect to the Inspektor Gadget nodes",
+			PossibleValues: []string{"kubeapi-server-per-node", "grpc-direct"},
+			DefaultValue:   "kubeapi-server-per-node",
+		},
+	}
```

Comment:
Either:
1. Validate somehow that it's set only to `kubeapi-server-per-node` 
2. Remove grpc-direct from allowed values

Reply:
Removed the whole param for now. We can add it back later on, when there are actually multiple connection methods available.